src/plugin.video.icdrama/lib/actions.py

```diff
@@ -169,7 +169,20 @@ def play_mirror(url):
     with common.busy_indicator():
         soup = BeautifulSoup(common.webread(url), 'html5lib')
         iframe = soup.find(id='iframeplayer')
-        iframe_url = urljoin(config.base_url, iframe.attrs['src'])
+        # decode added to fix 'Cannot mix str and non-str arguments' in Python 2
+        # try catch added to fix AttributeError: 'str' object has no attribute 'decode' in Python 3
+        # Probably one of the two is not str and we need to correct that
+        base_url = config.base_url
+        iframe_src = iframe.attrs['src']
+        try:
+            base_url = config.base_url.decode('utf-8')
+        except (UnicodeDecodeError, AttributeError):
+            pass
+        try:
+            iframe_src = iframe.attrs['src'].decode('utf-8')
+        except (UnicodeDecodeError, AttributeError):
+            pass
```

Comment:
You can replace lines 176 - 184 with just iframe_url = str(iframe.attrs['src']). That will turn it into/keep as a str object in Python 2/3, allowing urljoin to be used.

Haha I didn't know you can comment like this

Reply:
👍 Changes are made. I was too lazy to check where the problem was and Python is not my main programming language 😉